lib/libUPnP/Neptune/Source/System/Win32/NptWin32SerialPort.cpp

```diff
@@ -1,337 +1,334 @@
+            dcb.fOutxCtsFlow = TRUE;
+            dcb.fInX = dcb.fDsrSensitivity = FALSE;
+            dcb.fRtsControl = RTS_CONTROL_HANDSHAKE;
+            dcb.fDtrControl = DTR_CONTROL_DISABLE;
+            break;
+
+        case NPT_SERIAL_PORT_FLOW_CONTROL_XON_XOFF:
+            dcb.fOutX = TRUE;
+            dcb.fOutxCtsFlow = dcb.fOutxDsrFlow = FALSE;
+            dcb.fInX = TRUE;
+            dcb.fDsrSensitivity = FALSE;
+            dcb.fRtsControl = RTS_CONTROL_DISABLE;
+            dcb.fDtrControl = DTR_CONTROL_DISABLE;
+            break;
+    }
+    switch (parity) {
+        case NPT_SERIAL_PORT_PARITY_NONE: dcb.fParity = FALSE; dcb.Parity = NOPARITY; break;
+        case NPT_SERIAL_PORT_PARITY_EVEN: dcb.fParity = TRUE;  dcb.Parity = EVENPARITY; break;
+        case NPT_SERIAL_PORT_PARITY_ODD: dcb.fParity  = TRUE;  dcb.Parity = ODDPARITY; break;
+        case NPT_SERIAL_PORT_PARITY_MARK: dcb.fParity = TRUE;  dcb.Parity = MARKPARITY; break;
+    }
+    if (!SetCommState(handle, &dcb)) {
+        CloseHandle(handle);
+        return NPT_FAILURE;
+    }
+
+    // create a reference to the FILE object
+    m_HandleReference = new NPT_Win32HandleWrapper(handle);
+
+    return NPT_SUCCESS;
+}
+
+/*----------------------------------------------------------------------
+|   NPT_Win32SerialPort::Close
++---------------------------------------------------------------------*/
+NPT_Result
+NPT_Win32SerialPort::Close()
+{
+    // release the file reference
+    m_HandleReference = NULL;
+
+    return NPT_SUCCESS;
+}
+
+/*----------------------------------------------------------------------
+|   NPT_Win32SerialPort::GetInputStream
++---------------------------------------------------------------------*/
+NPT_Result 
+NPT_Win32SerialPort::GetInputStream(NPT_InputStreamReference& stream)
+{
+    // default value
+    stream = NULL;
+
+    // check that the file is open
+    if (m_HandleReference.IsNull()) return NPT_ERROR_SERIAL_PORT_NOT_OPEN;
+
+    // create a stream
+    stream = new NPT_Win32SerialPortInputStream(m_HandleReference);
+
+    return NPT_SUCCESS;
+}
+
+/*----------------------------------------------------------------------
+|   NPT_Win32SerialPort::GetOutputStream
++---------------------------------------------------------------------*/
+NPT_Result 
+NPT_Win32SerialPort::GetOutputStream(NPT_OutputStreamReference& stream)
+{
+    // default value
+    stream = NULL;
+
+    // check that the file is open
+    if (m_HandleReference.IsNull()) return NPT_ERROR_SERIAL_PORT_NOT_OPEN;
+
+    // create a stream
+    stream = new NPT_Win32SerialPortOutputStream(m_HandleReference);
+
+    return NPT_SUCCESS;
+}
+
+/*----------------------------------------------------------------------
+|   NPT_SerialPort::NPT_SerialPort
++---------------------------------------------------------------------*/
+NPT_SerialPort::NPT_SerialPort(const char* name)
+{
+    m_Delegate = new NPT_Win32SerialPort(name);
+}

```

Comment:
Something went wrong here. The whole file is shown as changed but you probably only changed a few lines.